src/evict/evict_lru.c

```diff
@@ -1906,7 +1906,8 @@ __evict_walk_file(WT_SESSION_IMPL *session,
 		mod = page->modify;
 		if (modified && txn_global->current != txn_global->oldest_id &&
 		    (mod->last_eviction_id == __wt_txn_oldest_id(session) ||
-		    !__wt_txn_visible_all(session, mod->update_txn, NULL)))
+		    !__wt_txn_visible_all(session, mod->update_txn,
+		    WT_TIMESTAMP_NULL(&mod->update_ts))))
```

Comment:
I thought this check was only supposed to be advisory -- i.e., it's only supposed to avoid wasted work.  Is it necessary for correctness?

There is a check that `mod->rec_max_timestamp` is `visible_all` in `__wt_page_can_evict` that I think of as the correctness check...